components/ntp_background_images/browser/ntp_sponsored_images_data.cc

```diff
@@ -324,6 +335,45 @@ base::Value NTPSponsoredImagesData::GetBackgroundAt(size_t campaign_index,
   return data;
 }
 
+base::Value NTPSponsoredImagesData::GetBackgroundByAdInfo(
+    const ads::NewTabPageAdInfo& ad_info) {
+  // Find campaign
+  size_t campaign_index = 0;
+  for (; campaign_index != campaigns.size(); ++campaign_index) {
+    if (campaigns[campaign_index].campaign_id == ad_info.campaign_id) {
+      break;
+    }
+  }
+  if (campaign_index == campaigns.size()) {
+    LOG(ERROR) << "Ad campaign wasn't found in NTP sposored images data: "
+               << ad_info.campaign_id;
+    return base::Value();
+  }
+
+  const auto& sponsored_backgrounds = campaigns[campaign_index].backgrounds;
+  size_t background_index = 0;
+  for (; background_index != sponsored_backgrounds.size(); ++background_index) {
+    if (sponsored_backgrounds[background_index].creative_instance_id ==
+        ad_info.creative_instance_id) {
+      break;
+    }
+  }
+  if (background_index == sponsored_backgrounds.size()) {
+    LOG(ERROR) << "Creative instance wasn't found in NTP sposored images data: "
+               << ad_info.creative_instance_id;
```

Comment:
nit: WARNING

Reply:
I think this should be an ERROR, because we are unable to show NTP SI due to difference between catalog and component updater metadata.

Reply:
Good point, thanks